pkg/secrets/push_to_file/secrets_groups.go

```diff
@@ -0,0 +1,235 @@
+)
+
+// SecretsPaths comprises Conjur variable paths for all secrets in a secrets
+// group, indexed by secret name.
+type SecretsPaths map[string]string
+
+// SecretsGroupInfo comprises secrets mapping information for a given secrets
+// group.
+type SecretsGroupInfo struct {
+	Secrets      SecretsPaths
+	FilePath     string
+	FileType     SecretsFileType
+	FilePerms    os.FileMode
+	FileTemplate *template.Template
+}
+
+// SecretsGroups comprises secrets mapping info for all secret groups
+type SecretsGroups map[string]SecretsGroupInfo
+
+func ExtractSecretsGroupsFromAnnotations(annotations map[string]string) (SecretsGroups, error) {
+	secretsGroups := make(SecretsGroups)
+
+	for annotation := range annotations {
+		if strings.HasPrefix(annotation, SECRETS_GROUP_PREFIX) {
+			groupName := strings.TrimPrefix(annotation, SECRETS_GROUP_PREFIX)
+
+			secretsPathPrefix, err := parseSecretsPathPrefix(groupName, annotations[SECRETS_GROUP_POLICY_PATH_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			secrets, err := parseSecretsPaths(groupName, annotations[SECRETS_GROUP_PREFIX+groupName], secretsPathPrefix)
+			if err != nil {
+				return nil, err
+			}
+
+			filePath, err := parseFilePath(groupName, annotations[SECRETS_GROUP_FILE_PATH_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			fileType, err := parseFileType(groupName, annotations[SECRETS_GROUP_FILE_TYPE_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			fileTemplate, err := parseFileTemplate(groupName, annotations[SECRETS_GROUP_FILE_TEMPLATE_PREFIX+groupName])
+			if err != nil {
+				return nil, err
+			}
+
+			if fileTemplate != nil {
+				fileType = FILE_TYPE_NONE
+			}
+
+			groupInfo := SecretsGroupInfo{
+				Secrets:      secrets,
+				FilePath:     filePath,
+				FileType:     fileType,
+				FilePerms:    DEFAULT_FILE_PERMS,
+				FileTemplate: fileTemplate,
+			}
+
+			err = validateGroupInfo(groupName, groupInfo)
+			if err != nil {
+				return nil, err
+			}
+
+			/*
+				if groupInfo.FileTemplate != nil {
+					groupInfo.FileTemplate.Execute(os.Stdout, groupInfo.Secrets)
+				}
+			*/
+
+			secretsGroups[groupName] = groupInfo
+		}
+	}
+
+	return secretsGroups, nil
+}
+
+func parseSecretsPaths(groupName string, secretsPaths string, secretsPathsPrefix string) (SecretsPaths, error) {
+	secrets := make(SecretsPaths)
+
+	unpacked := []interface{}{}
+	if err := yaml.Unmarshal([]byte(secretsPaths), &unpacked); err != nil {
+		return nil, log.RecordedError(messages.CSPFK051E, err.Error(), groupName)
+	}
+
+	insertSecret := func(name string, secretPath string) error {
+		secretPath = path.Clean(secretPath)
+		if secretPath == "." {
+			return log.RecordedError(messages.CSPFK051E, "Invalid secret path", groupName)
+		}
+		secrets[name] = secretsPathsPrefix + secretPath
+
+		return nil
+	}
+
+	for _, secret := range unpacked {
+		switch val := secret.(type) {
+		case string:
+			{
+				name := val[strings.LastIndex(val, "/")+1:]
```

Comment:
Maybe using `path.Base()` would be simpler for extracting the last word of the variable path?

Reply:
Done